pkg/deploy/gateway/gateway.go

```diff
@@ -318,7 +377,71 @@ func getGatewayRoleBindingSpec(instance *orgv1.CheCluster) rbac.RoleBinding {
 	}
 }
 
+func getGatewayOauthProxyConfigSpec(instance *orgv1.CheCluster, secret *corev1.Secret) corev1.ConfigMap {
+	cookieSecret := secret.Data["cookie_secret"]
```

Comment:
Why not to pass the content of the secret directly as an argument?

Reply:
fixed